### PR TITLE
[Merged by Bors] - feat: add deduplication doc

### PR DESCRIPTION
**Pull request by ozgrakkurt**
Added small document that explains how deduplication works

### CODE REVIEW

File: content/smartmodules/deduplication.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+```bash
+$ fluvio topic create -c topic.yaml
+```
+
+## Parameters
+
+| Parameter       | default | type   | optional | description                                           |
+|:-------------|:--------| :---   | :---   |:------------------------------------------------------|
+| count |    -    | Integer | false | Minimum number of records the filter will remember. It doesn't guarantee to remember records that came `count` records before now. |
+| age   |    -    | Integer | true | Minimum amount of time this filter will remember a record for. It can be specified using this format: `15days 2min 2s`, or `2min 5s`, or `15ms` | |
```
Comment (ajhunyady) [inline content/smartmodules/deduplication.md:41]: Please provide an example:
1. Produce
2. Consume
3. Show how items are deduplicated

Explain how it works, meaning that deduplication uses record key.

**Review by ajhunyady — CHANGES_REQUESTED** [review]

File: content/smartmodules/deduplication.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+title: Deduplication
+---
+
+The `Deduplication` feature in Fluvio is the way to deduplicate records based on their keys.
+
+Deduplication can be used on a topic directly, or it can be used on connectors. It will drop the duplicate records on a configured window.
```
[the remark this answers is not in the document]
Comment (ozgrakkurt) [reply content/smartmodules/deduplication.md:7]: Not sure about this so left it vague, I'll need to check or ask Alexander when he is back

[the remark this answers is not in the document]
Comment (ajhunyady) [reply content/smartmodules/deduplication.md:7]: I don't think it's supported in connectors, and I don't think it's needed. 
The deduplication needs to be close to the stream as it needs to re-scan, that's why we applied to topic.

[the remark this answers is not in the document]
Comment (ozgrakkurt) [reply content/smartmodules/deduplication.md]: removed mention of connector config

File: content/smartmodules/deduplication.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+This window is defined by the parameters passed to the filter smartmodule.
```
[the remark this answers is not in the document]
Comment (ozgrakkurt) [reply content/smartmodules/deduplication.md]: I made a change here, can you re-check?

File: content/smartmodules/deduplication.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+---
+title: Deduplication
+---
+
+The `Deduplication` feature in Fluvio is the way to deduplicate records based on their keys.
```
[the remark this answers is not in the document]
Comment (ozgrakkurt) [reply content/smartmodules/deduplication.md:5]: Added more info about the behavior and also some implementation details.

File: content/smartmodules/deduplication.md (hunk 1 above)
Comment (ozgrakkurt) [reply content/smartmodules/deduplication.md:63]: Added simple example from the notion document

Comment (ajhunyady) [reply content/smartmodules/deduplication.md:63]: Nice work!

**Review by ajhunyady — APPROVED** [review]
Good work!

**Comment by ozgrakkurt** [comment]
> Good work!

thanks!

**Pull request closed without merging**